app/jobs/two_factor_authentication_removal_job.rb

```diff
@@ -32,8 +32,7 @@ def perform
         # Return publisher to previous state.
         ActiveRecord::Base.transaction do
           if publisher.locked?
-            # Status_updates is sorted by entry, first is "locked", second is the status the publisher was in before
-            previous_status = publisher.status_updates&.second&.status || PublisherStatusUpdate::ACTIVE
+            previous_status = publisher.status_updates.select { |status_update| status_update.status != PublisherStatusUpdate::LOCKED }.first.status
```

Comment:
Was going to comment that we should keep the default in case there is a missing previous status, but if you're locked you must have had a previous one. If you don't, that's a bug and the data and code should be fixed.

Reply:
There's always a previous status.